lib/puppet/reports/http.rb

```diff
@@ -14,7 +14,7 @@
   def process
     url = URI.parse(Puppet[:reporturl])
     headers = { "Content-Type" => "application/x-yaml" }
-    options = {}
+    options = { :metric_id => [:puppet, :report, :http] }
```

Comment:
> We shouldn't ever end up with the case where elements passed in on the options hash are opaquely translated into content in the request, though, should we - e.g., as a query param on the URL or multipart form data? My understanding has been that the options hash is a mechanism for passing through settings which must be explicitly read/acted upon by the connection class.

Oh right. Good point.

> Embedding knowledge about the specific implementation class used in code that lives outside of the Puppet repo would seem dangerous - e.g., in the event that a future refactor were to change the name of the default implementation class which gets used

Agreed 100%. Even as I was writing that I was skeptical of it. The root concern is it just seems odd to add the facility into puppet when its not used/understood by puppet - eg it loosely feels like inappropriate knowledge, just going in the other direction (server => puppet). That said I'm not sure of a cleaner way and as you suggest `:metric_id` may be of expanded use in the future. I'm good with just a code comment explaining the use/background.